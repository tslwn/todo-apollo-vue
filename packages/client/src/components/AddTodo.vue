<template>
  <v-text-field
    class="px-4"
    placeholder="Add todo"
    v-model="text"
    v-on:keyup.enter="onEnter"
  ></v-text-field>
</template>

<script lang="ts">
import Vue from 'vue';
import { addTodoMutation, addTodoUpdate } from '../graphql/addTodo.mutation';

export default Vue.extend({
  name: 'add-todo',
  data: () => ({
    text: '',
  }),
  methods: {
    onEnter() {
      const { text } = this;

      this.text = '';

      this.$apollo
        .mutate({
          mutation: addTodoMutation,
          variables: {
            text,
          },
          update: addTodoUpdate,
          optimisticResponse: {
            __typename: 'Mutation',
            addTodo: {
              __typename: 'TodoUpdateResponse',
              success: true,
              message: 'Todo added successfully',
              todo: {
                __typename: 'Todo',
                // Mock ID
                id: -1,
                text,
                isComplete: false,
                isArchived: false,
              },
            },
          },
        })
        .catch(() => {
          // restore user input
          this.text = text;
        });
    },
  },
});
</script>

<style scoped></style>
