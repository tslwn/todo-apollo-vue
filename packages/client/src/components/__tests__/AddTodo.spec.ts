import Vue from 'vue';
import Vuetify from 'vuetify';
import { mount, createLocalVue } from '@vue/test-utils';
import AddTodo from '../AddTodo.vue';

Vue.use(Vuetify);

const localVue = createLocalVue();

describe('AddTodo.vue', () => {
  let vuetify: typeof Vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('renders correctly', () => {
    const wrapper = mount(AddTodo, {
      localVue,
      vuetify,
      propsData: { text: 'Add unit tests' },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
