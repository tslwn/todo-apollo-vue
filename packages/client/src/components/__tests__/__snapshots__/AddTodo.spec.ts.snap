// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo.vue renders correctly with text 1`] = `
<div class="v-input px-4 theme--light v-text-field v-text-field--placeholder">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-text-field__slot"><input id="input-12" placeholder="Add todo" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddTodo.vue renders correctly without text 1`] = `
<div class="v-input px-4 theme--light v-text-field v-text-field--placeholder">
  <div class="v-input__control">
    <div class="v-input__slot">
      <div class="v-text-field__slot"><input id="input-7" placeholder="Add todo" type="text"></div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
